draft/XXXX-put-row-endpoint/README.md

```diff
@@ -0,0 +1,97 @@
+---
+RFC: XXXX
+Author: Roni Choudhury
+Created: 2021-01-26
+Last-Modified: 2021-01-26
+---
+
+# RFC XXXX: PUT endpoint for appending/modifying data in a table
+
+This RFC proposes a new `PUT` endpoint for table data, enabling appending of new
+rows, and modification of rows already in the table.
+
+## Motivation
+
+Up until now, Multinet's data model has been *static*; that is, once data is
+uploaded to the system, it never changes. This property has some nice effects on
+system design, especially for envisioned (but, as of now non-existent) features
+such as canned views of specific tables, or of certain network operations based
+on static underlying table data.
+
+However, we now have a specific use case for which dynamic data is necessary.
+Because the predicted benefits of a purely static data model never materialized,
+we are relatively free to make this change now to address actual needs that have
+arisen.
+
+The use case is from the Marclab, where data collection is a long, ongoing
+process. There are two modes of data update the lab needs:
+
+1. Sometimes, new data is computed and the appropriate rows must simply be
+   appended to an existing data table. This is the case of conceptually
+   incomplete data following a gradual procedure of completion.
+2. Other times, existing data must be updated due to new knowledge discovery.
+   This is a case of conceptually incorrect data undergoing a process of
+   correction.
+
+Finally, there is a third ancillary use case solved by the `PUT` endpoint: our
+infrastructure can currently only support a maximum time of 30 seconds per
+request, which is not long enough to support the uploading of very large
+datasets (on the order of a 30MB CSV file). Though an in-flight RFC is
+addressing this problem organically, the `PUT` endpoint nonetheless provides a
+workaround: simply slice the CSV file into smaller tranches of data, then use
+the `PUT` endpoint to append these tranches to reconstruct the whole table.
+
+## Proposals
+
+### `PUT` endpoint
+
+We are proposing to add a new endpoint:
+
+```
+PUT /api/workspaces/{workspace}/tables/{table}
```

Comment:
If the other endpoints (including your structure change section) use the `rows` suffix, shouldn't this as well?


```suggestion
PUT /api/workspaces/{workspace}/tables/{table}/rows
```

Reply:
I'm usually a proponent of smaller changes done in sequence, but you are right, if we don't make your suggested change, we'll end up with an inconsistent API until we apply the fix in the structure section.

So what the hell, let's go ahead and make that change now.